crates/core/src/network/types.rs

```diff
@@ -421,7 +421,7 @@ impl<M: Targeted + WireEncode> Outgoing<M, HasConnection> {
         let connection = bail_on_error!(self, self.try_upgrade());
         let permit = bail_on_none!(
             self,
-            connection.reserve().await,
+            tokio::task::unconstrained(connection.reserve()).await,
```

Comment:
Did you observe an effect of cooperative scheduling w/o `unconstrained`?

Reply:
Yes, although it of course varies depending on the workload. I was able to observe 30% improvement in tail latencies of GetSequencerState processing when the system is under heavy load.